packages/vscode-dependencies-validation/src/autofix/configuration.ts

```diff
@@ -0,0 +1,18 @@
+import { VscodeWorkspace } from "../vscodeTypes";
+
+export const ENABLE_AUTOFIX = "dependenciesValidation.enableAutoFix";
+const DELAY_AUTOFIX = "dependenciesValidation.delayAutoFix";
+
+export function isAutoFixEnabled(workspace: VscodeWorkspace): boolean {
```

Comment:
I think a **defaults** object which would allow only a single getConfig method.
could make this code cleaner (assuming there would be more user settings in the future).

I will try to implement this, there could be issues with the type information though...

Reply:
WDYT?

```typescript
import { workspace } from "vscode"; // change this to avoid direct dep on vscode
type VscodeWorkspace = typeof workspace

export const ENABLE_AUTOFIX = "dependenciesValidation.enableAutoFix";
export const DELAY_AUTOFIX = "dependenciesValidation.delayAutoFix";

interface ExtensionConfiguration {
    [ENABLE_AUTOFIX] : boolean
    [DELAY_AUTOFIX] : number
}

type ExtConfigKey = keyof  ExtensionConfiguration

const ExtensionConfigDefaults: ExtensionConfiguration = {
    [ENABLE_AUTOFIX] : false,
    [DELAY_AUTOFIX] : 0
}

export function getConfigValue<T extends ExtConfigKey>(key: T, workspace:VscodeWorkspace): ExtensionConfiguration[T] {
    const workspaceConfig = workspace.getConfiguration();
    const value =  workspaceConfig.get(key, ExtensionConfigDefaults[key])
    return value;
}


```